### PR TITLE
New pull request template

### LINKED ISSUE #98: Addition of Pull request template
## 📃 Summary

Community profile suggests the use of a Pull request template

## Expected documentation

Pull request template should exist and provide value to the PR process

### PR DESCRIPTION
**Pull request by mxhenry-godaddy**
## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Tests
[*] Other
```
## Breaking Change?
```
[ ] Yes
[*] No
```

## Issue Linking
closes #98 

## What's new?
Adds a pull request template to .github/

### CODE REVIEW

**Comment by tarkatronic** [comment]
@mxhenry-godaddy Thanks for starting this work!

I'd like to get a look at some pre-existing PR templates from around the community, similar to what I did initially for the issue templates. One thing I've seen in the past that I particularly like is having a checklist for contributors. Something like:

> To help us get this pull request reviewed and merged quickly, please be sure to include the following items:
> 
> * [ ] Tests (if applicable)
> * [ ] Documentation (if applicable)
> * [ ] Changelog entry
> * [ ] A full explanation here in the PR description of the work done
> * [ ] Mention any relevant issue(s) with [closing keywords](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

^ Those are just things off the top of my head. I'm sure I'm missing things. And the wording can probably use some massaging. That's why I like looking for pre-existing examples. 😄

**Comment by mxhenry-godaddy** [comment]
Examples I could find that felt like good jump-off points:
[vgpena/gh-docs-boilerplate](https://github.com/vgpena/gh-docs-boilerplate/blob/master/.github/pull_request_template.md)
[Pull Request Templates on GitHub](https://tech.boldare.com/pull-request-templates-on-github/)

Also, I'm doing a dive on [What every reviewer would like to see in your next Pull Request](https://nebulab.it/blog/what-every-reviewer-would-like-to-see-in-your-next-pull-request/) for inspiration, especially the use of multiple PR templates customized to the change type

**Comment by mxhenry-godaddy** [comment]
@tarkatronic I'd like to move forward on this if possible and iterate once contributors start using it

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+To help us get this pull request reviewed and merged quickly, please be sure to include the following items:
+```
+[ ] Tests (if applicable)
+[ ] Documentation (if applicable)
+[ ] Changelog entry
+[ ] A full explanation here in the PR description of the work done
+```
```
Comment (tarkatronic) [inline .github/pull_request_template.md]:
```suggestion

* [ ] Tests (if applicable)
* [ ] Documentation (if applicable)
* [ ] Changelog entry
* [ ] A full explanation here in the PR description of the work done

```
If you leave them out of the code fencing and make them a list, GitHub should actually turn these into checkboxes for the user to check. :)

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+[ ] Bugfix
+[ ] Feature
+[ ] Code style update (formatting, local variables)
+[ ] Refactoring (no functional changes, no api changes)
+[ ] Build related changes
+[ ] CI related changes
+[ ] Documentation content changes
+[ ] Tests
+[ ] Other
+```
```
Comment (tarkatronic) [inline .github/pull_request_template.md]:
```suggestion

* [ ] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [ ] Tests
* [ ] Other

```

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+## Breaking Change?
+```
+[ ] Yes
+[ ] No
+```
```
Comment (tarkatronic) [inline .github/pull_request_template.md]:
```suggestion
## Backward Compatibility

Is this change backward compatible with the most recently released version? Does it introduce changes which might change the user experience in any way? Does it alter the API in any way?

[ ] Yes (backward compatible)
[ ] No (breaking changes)

```

**Review by tarkatronic — CHANGES_REQUESTED** [review]

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+* [ ] Documentation content changes
+* [ ] Tests
+* [ ] Other
+
+## Backward Compatibility
+
+Is this change backward compatible with the most recently released version? Does it introduce changes which might change the user experience in any way? Does it alter the API in any way?
+
+[ ] Yes (backward compatible)
+[ ] No (breaking changes)
```
Comment (tarkatronic) [inline .github/pull_request_template.md]: @mxhenry-godaddy My bad, I missed the list markers on these two. Last change, I swear! 😄 
```suggestion
* [ ] Yes (backward compatible)
* [ ] No (breaking changes)
```

**Review by sjacoby-godaddy — APPROVED** [review]

**Review by tarkatronic — APPROVED** [review]

**Pull request merged**